README.md

```diff
@@ -93,9 +93,129 @@ TODO aspect ratio + images alignment + more more images
+```c++
+// This doesn't actually free the memory from the pool, the memory is queued up to be freed only after the `scratchSemaphore` reaches a value a future submit will signal
+memory_pool->deallocate(&offset,&size,nextSubmit.getFutureScratchSemaphore());
+```
+
+### 🧩 GPU Object Lifecycle Tracking
+
+Nabla uses [smart reference counting]() to track the lifecycle of GPU objects. Descriptor sets and command buffers are responsible for maintaining reference counts on the resources (e.g., buffers, textures) they use. The queue itself also tracks command buffers, ensuring that objects remain alive as long as they are pending execution. This system guarantees the correct order of deletion and makes it difficult for GPU objects to go out of scope and be destroyed before the GPU has finished using them.
+
+### 🧩 HLSL2021 Standard Template Library
+
+- 🔄 Reusable: Unified single-source C++/HLSL libraries eliminate code duplication with reimplementation of STL's `type_traits`, `limits`, `functional`, `tgmath`, etc.
+
+- 🐞 Shader Logic, CPU-Tested: A subset of HLSL compiles as both C++ and SPIR-V, enabling CPU-side debugging of GPU logic, ensuring correctness in complex tasks like FFT, Prefix Sum, etc.
+Future Proof: C++20 Concepts in HLSL for safe and documented Static Polymorphism
+
+- 🔮 Future-Proof: C++20 concepts in HLSL enable safe and documented polymorphism.
+
+- 🧠 Insane: Boost Preprocessor and Template Metaprogramming in HLSL!
+
+- 🛠️ Real-World Problem Solvers: The library offers GPU-optimized solutions for tasks like Prefix Sum, Binary Search, FFT, Global Sort, and even emulated `shaderFloat64` when native GPU support is unavailable!
+
+```
+[TODO][CODE] Code for each or just one showcasing most of the above points?
+```
+
+### 🧩 Full Embrace of [Buffer Device Address]() and [Descriptor Indexing]()
+
+By utilizing Buffer Device Addresses (BDAs), Nabla enables more direct access to memory through 64-bit GPU virtual addresses. Synergized with Descriptor Indexing, this approach enhances flexibility by enabling more dynamic, scalable resource binding without relying on traditional descriptor sets.
+
+### 🧩 Minimally Invasive Design
+[TODO]: vulkan handle acquisition, multiple windows, content playing second fiddle
+
+### 🧩 Designed for Interoperation
+Nabla is built with interoperation in mind, supporting memory export and import between different compute and graphics APIs.
+
+----
+
+[TODO]:
+- Cancellable Future based Async I/O
+- Virtual File System (archive mounting, our alternative to #embed, everything is referenced by absolute - path)
+
+----
+
+### 🧩 Asset Manager
+Nabla’s Asset Manager efficiently loads assets while tracking dependencies using a Directed Acyclic Graph (DAG). Assets are loaded in the correct order, avoids redundant allocations, and simplifies resource management.
```

Comment:
rewrite, its missing the point. Alos not sure Asset Manager is something to boast about. Its the Asset System/Design that matters, the manager is actually quite crap.

Reply:
> Asset Managment: Directed Acyclic Graphs
Tried to fill in something for the point in you gist with little too no knowledge of what it meant xD. That's on me. Can you tell me which system you're referring to exactly?

Reply:
the fact that ICPU mirror IGPU and there's a 1:1 mapping there, also the loaders for different asset types can invoke each other so OBJ invokes MTL and MTL invokes image loaders